src/vasim/simulator/InMemorySimulator.py

```diff
@@ -130,14 +166,11 @@ def _initialize_output_file(data_dir):
         out_filename = f"{data_dir}/decisions.csv"
         out_file = Path(out_filename)
 
+        # Use 'with' statement to open the file
         if not out_file.exists():
-            f = open(out_file, "a", encoding="utf-8")
-            f.write("LATEST_TIME,CURR_LIMIT,NEW_LIMIT\n")
-            f.flush()
-        else:
-            f = open(out_file, "a", encoding="utf-8")
-
-        return f
+            with open(out_file, "a", encoding="utf-8") as f:
+                f.write("LATEST_TIME,CURR_LIMIT,NEW_LIMIT\n")
+        return open(out_file, "a", encoding="utf-8")
```

Comment:
I'm not certain this really solves the fact that we return an open file handle and never close it.

One way you could work around this though is as follows (basically a ContextManager pattern):

```suggestion
        with open(out_file, "a", encoding="utf-8") as f:
            yield f
```